### DISCUSSION TITLE
How to start with custom native plugin

### DISCUSSION THREAD

**Discussion by bemon** [osmandapp/OsmAnd#12281] [Q&A]
Hi, 
I'm a developer and cyclist. I wonder how to begin writing custom plugin to show heart rate sensor ( i have polar h10 that provides native android and ios SDK  to read data). I could't find any materials on this topic. Is it possible to write osm plugin and display some data on top of map ( like already avaible speed, battery charge etc ). Can someone provide any directions how to start / where to look ?

**Comment by vshcherb** [comment]
You probably could start with samples repo https://github.com/osmandapp/osmand-api-demo/tree/master/OsmAnd-api-sample and small description https://osmand.net/build_it.

Unfortunately documentation for API is not complete either https://docs.osmand.net/en/main@latest/development/build-osmand/osmand-api
